workflow/scripts/parallel_vcf.py

```diff
@@ -0,0 +1,50 @@
+import os
+import argparse
+from multiprocessing import Pool
+from pysam import VariantFile
+import tempfile
+import sys
+
+parser = argparse.ArgumentParser()
+parser.add_argument("file_vcf")
+parser.add_argument("command")
+parser.add_argument("--stepsize", "-s", type=int, default=1000000)
+parser.add_argument("--threads", "-t", type=int, default=8)
+parser.add_argument("--tmp", default=None, help="Define the base directory for the tmp files")
+args = parser.parse_args()
+tmp_base = args.tmp
+stepsize = args.stepsize
+
+# create regions
+f = VariantFile(args.file_vcf)
+regions = []
+for name, contig in f.header.contigs.items():
+    last = 0
+    for x in list(range(0, contig.length, stepsize)):
+        stop = min(last + stepsize - 1, contig.length)
+        regions.append(f"{name}:{last}-{stop}")
+        last = stop+1
+f.close() 
+
+#command = "SnpSift annotate {ss_args} {args.database_vcf} /dev/stdin"
+command = args.command
+
+with tempfile.TemporaryDirectory(dir=tmp_base) as t:
+    def f(x):
+        #print("start", x, file=sys.stderr)
+        cmd = f"bcftools view -r {x} {args.file_vcf} | bcftools view -t {x} | {command} | bcftools view -Ob > {t}/{x}.bcf"
```

Comment:
* Why two views behind each other?
* Whenever piping between bcftools, one should use -Ou instead of VCF output (uncompressed binary instead of plain text).

Reply:
The reason for the 2 views is rather stupid. When you pass a region r to bcftools and uses the indexed view, then any variant overlapping the start position of r is taken. So even Indels that start outside of r. Of cause, then you have some duplicated variants, if you split up the vcf like I do and there was no flag to prevent this. This is not the case if you use the streaming view with a region (-t). But on the downside the streaming view does not jump right to the region position, but goes through everything before. Therefore the easiest way I could imaging was to first use the indexed view and then eliminating variants, that starts before r by using the streaming view.